docs/architecture_overview.md

```diff
@@ -0,0 +1,130 @@
+
+
+
+Each of these classes can be considered themselves as generalizations which can be configured for usage in specific instances. As such, each of these classes provides a list of parameters that are used to configure their behaviors, and subclasses can provide additional parameters for additional specification. No `Blueprint` should only work for one specific tasks, but it should be a blueprint from which many similar tasks can be built.
+
+The classes are all expected to provide an `Args` class, which is a dataclass that we can use as a structured config alongside [Hydra](https://hydra.cc/) to populate configuration. It should also define an `assert_task_args` method, which ensures that given a set of configuration, a run with that particular configuration will not fail to execute for any configuration related reasons.
+
+There's also the abstraction of the `MephistoDB`, which defines the data operations that Mephisto requires to operate properly. If a specific setup requires specialized data handling, any class implementing this interface could stand in for it. 
```

Comment:
Should we also list MephistoDB as a core abstraction (making the 3 abstractions into 4?). It's mentioned in the abstractions section here so just wondering.

Reply:
It's hard to know - I wouldn't say that the `MephistoDB` is a clean enough abstraction for me to list it as such, and further it's not one of the ones that Mephisto knows how to auto-populate. Much of mephisto code currently depends on the LocalMephistoDB specifically.

Once I get a chance to clean up that abstraction, I'll likely list it directly with the rest.